dashboard/app/controllers/scripts_controller.rb

```diff
@@ -26,7 +26,9 @@ def show
     end
 
     if !params[:section_id] && current_user&.last_section_id
-      redirect_to "#{request.path}?section_id=#{current_user.last_section_id}"
+      query_string = request.query_string.empty? ? '' : "#{request.query_string}&"
+      section_id_query_string = "section_id=#{current_user.last_section_id}"
+      redirect_to "#{request.path}?" + query_string + section_id_query_string
```

Comment:
I think redirect_to can take a hash of query params.  Would that be clearer than trying to build the query string by hand?

Reply:
I actually tried that, by merging section_id into the existing params, but then got some kind of param not allowed error.. I'll try again.

Reply:
That's weird.  Is there a strong params thing that we need to update somewhere?